diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/FileStructureRule.kt

```diff
@@ -245,14 +246,19 @@ class FileStructureRule(configRules: List<RulesConfig>) : DiktatRule("file-struc
     }
 
     private fun findAllReferences(node: ASTNode) {
-        node.findAllNodesWithSpecificType(OPERATION_REFERENCE)?.forEach { ref ->
+        node.findAllNodesWithSpecificType(OPERATION_REFERENCE).forEach { ref ->
             if (!ref.isPartOf(IMPORT_DIRECTIVE)) {
-                operatorMap.filterValues { it == ref.text }.keys.forEach { key -> refSet.add(key) }
+                val references = operatorMap.filterValues { it == ref.text }
+                if (references.isNotEmpty()) {
+                    references.keys.forEach { key -> refSet.add(key) }
+                } else {
+                    refSet.add(ref.text)
```

Comment:
Why is this branch needed? Does it catch infix functions? Please add a comment in code

Reply:
Yes. This is needed to check infix  functions that relate to OPERATION_REFERENCE